webkit2/jscore.js-value.lisp

```diff
@@ -0,0 +1,156 @@
+(export 'jsc-value-to-int32)
+
+(defcfun "jsc_value_new_boolean" (g-object jsc-value)
+  (context (g-object jsc-context)))
+(export 'jsc-value-new-boolean)
+
+(defcfun "jsc_value_is_boolean" :boolean
+  (value (g-object jsc-value)))
+(export 'jsc-value-is-boolean)
+
+(defcfun "jsc_value_to_boolean" :boolean
+  (value (g-object jsc-value)))
+(export 'jsc-value-to-boolean)
+
+(defcfun "jsc_value_new_string" (g-object jsc-value)
+  (value (g-object jsc-value)))
+(export 'jsc-value-new-string)
+
+;; TODO: jsc_value_new_string_from_bytes
+
+(defcfun "jsc_value_is_string" :boolean
+  (value (g-object jsc-value)))
+(export 'jsc-value-is-string)
+
+(defcfun "jsc_value_to_string" :string
+  (value (g-object jsc-value)))
+(export 'jsc-value-to-string)
+
+;; TODO: jsc_value_new_array
+
+(defcfun "jsc_value_is_array" :boolean
+  (value (g-object jsc-value)))
+(export 'jsc-value-is-array)
+
+(defcfun "jsc_value_new_object" (g-object jsc-value)
+  (context (g-object jsc-context))
+  (instance :pointer)
+  (class (g-object jsc-class)))
+(export 'jsc-value-new-object)
+
+(defcfun "jsc_value_is_object" :boolean
+  (value (g-object jsc-value)))
+(export 'jsc-value-is-object)
+
+(defcfun "jsc_value_object_is_instance_of" :boolean
+  (value (g-object jsc-value))
+  (name :string))
+(export 'jsc_value_object-is-instance-of)
+
+(defcfun "jsc_value_object_set_property" :void
+  (value (g-object jsc-value))
+  (name :string)
+  (property (g-object jsc-value)))
+(export 'jsc_value-object-set-property)
+
+(defcfun "jsc_value_object_get_property" (g-object jsc-value)
+  (value (g-object jsc-value))
+  (name :string))
+(export 'jsc_value-object-get-property)
+
+(defcfun "jsc_value_object_set_property_at_index" :void
+  (value (g-object jsc-value))
+  (index :int)
+  (property (g-object jsc-value)))
+(export 'jsc_value_object_set-property-at-index)
+
+(defcfun "jsc_value_object_get_property_at_index" (g-object jsc-value)
+  (value (g-object jsc-value))
+  (index :int))
+(export 'jsc_value_object_get-property-at-index)
+
+(defcfun "jsc_value_object_has_property" :boolean
+  (value (g-object jsc-value))
+  (name :string))
+(export 'jsc_value-object-has-property)
+
+(defcfun "jsc_value_object_delete_property" :boolean
+  (value (g-object jsc-value))
+  (name :string))
+(export 'jsc_value-object-delete-property)
+
+(defcfun "jsc_value_object_enumerate_properties" (:pointer :string)
+  (value (g-object jsc-value)))
+(export 'jsc_value-object-enumerate-properties)
+
+;; TODO:
+;; JSCValue *	jsc_value_object_invoke_method ()
+;; JSCValue *	jsc_value_object_invoke_methodv ()
+;; void	jsc_value_object_define_property_data ()
+;; void	jsc_value_object_define_property_accessor ()
+;; JSCValue *	jsc_value_new_function ()
+;; JSCValue *	jsc_value_new_functionv ()
+;; JSCValue *	jsc_value_new_function_variadic ()
+;; JSCValue *	jsc_value_function_call ()
+;; JSCValue *	jsc_value_function_callv ()
+;; JSCValue *	jsc_value_constructor_call ()
+;; JSCValue *	jsc_value_constructor_callv ()
```

Comment:
Is there a particular blocker for these, or did you just grow tired? :)

Reply:
I haven't seen any particular use for these in Lisp land, and they had quite a complex vararg list, so yeah, I've gotten lazy about implementing it :)

Maybe later, if we need them.